framework_tvb/tvb/core/services/simulator_service.py

```diff
@@ -226,13 +237,200 @@ def async_launch_and_prepare_pse(self, burst_config, user, project, simulator_al
+        form = SimulatorRMFragment(surface_index, simulator.connectivity)
+        rm_fragment = self.algorithm_service.prepare_adapter_form(form_instance=form, project_id=project_id)
+        rm_fragment.fill_from_trait(simulator.surface)
+
+        rendering_rules.form = rm_fragment
+        rendering_rules.form_action_url = form_action_url
+        return rendering_rules.to_dict()
+
+    def prepare_stimulus_fragment(self, simulator, rendering_rules, is_surface_simulation, form_action_url, project_id):
+        form = SimulatorStimulusFragment(is_surface_simulation)
+        stimuli_fragment = self.algorithm_service.prepare_adapter_form(form_instance=form, project_id=project_id)
+        stimuli_fragment.fill_from_trait(simulator)
+
+        rendering_rules.form = stimuli_fragment
+        rendering_rules.form_action_url = form_action_url
+        return rendering_rules.to_dict()
+
+    def prepare_next_fragment_after_noise(self, simulator, is_branch, rendering_rules):
+        if isinstance(simulator.integrator.noise, AdditiveNoiseViewModel):
+            rendering_rules.is_branch = is_branch
+            return self.prepare_monitor_fragment(simulator, rendering_rules, SimulatorWizzardURLs.SET_MONITORS_URL)
+
+        equation_form = get_form_for_equation(type(simulator.integrator.noise.b))()
+        equation_form.equation.data = simulator.integrator.noise.b.equation
+        equation_form.fill_from_trait(simulator.integrator.noise.b)
+
+        rendering_rules.form = equation_form
+        rendering_rules.form_action_url = SimulatorWizzardURLs.SET_NOISE_EQUATION_PARAMS_URL
+        return rendering_rules.to_dict()
+
+    @staticmethod
+    def prepare_monitor_fragment(simulator, rendering_rules, form_action_url):
+        monitor_fragment = SimulatorMonitorFragment(simulator.is_surface_simulation)
+        monitor_fragment.fill_from_trait(simulator.monitors)
+
+        rendering_rules.form = monitor_fragment
+        rendering_rules.form_action_url = form_action_url
+        return rendering_rules.to_dict()
+
+    def build_list_of_monitors_from_names(self, monitor_names, is_surface_simulation):
+        self.monitors_dict = dict()
+        monitor_dict = get_ui_name_to_monitor_dict(is_surface_simulation)
+
+        count = 0
+        for monitor_name in monitor_names:
+            monitor = monitor_dict[monitor_name]
+            self.monitors_dict[monitor.__name__] = (monitor(), count + 1)
+            count = count + 1
+
+        return list(monitor[0] for monitor in self.monitors_dict.values())
+
+    def build_list_of_monitors_from_view_models(self, monitor_view_models):
+        self.monitors_dict = dict()
+        count = 0
+        for monitor_model in monitor_view_models:
+            self.monitors_dict[monitor_model.__class__.__name__] = (monitor_model, count + 1)
+            count = count + 1
+
+    @staticmethod
+    def build_monitor_url(fragment_url, monitor):
```

Comment:
URL build in the service does not seem right. Should be in the controllers, because there we have the code responding to such urls, isn't it?